### DISCUSSION TITLE
Branch Protection

### DISCUSSION THREAD

**Discussion by timbray** [timbray/quamina#325] [Q&A]
I just turned off review-approvals-required for Quamina, because at the moment, we lack reviewers.

After a long time of being busy with other stuff, I have time for Open Source again, and Quamina is my summer programming project.  I suspect that if I hadn't stalled for 18 months, Quamina would still have an engaged community of people who had some combination of wanting to use it for things or being interested in finite automata and were happy to review.  But I do acknowledge that wrestling with NFAs is fairly taxing stuff and a lot of people don't find it fun.

We have a long list of interesting features and, now that the exploding-NFA roadblock has been removed, I'm eager to enhance Quamina in a bunch of directions, and work towards full compatibility with Ruler.  So I plan to be doing PRs regularly for the next few months, quite likely one per week.

I am quite happy with the testing and validation coverage so the risk of adding features is pretty low, especially since most of the features are precisely defined by their Ruler equivalents and there are high-quality unit tests to be stolen from Ruler.

I know that reviews add value and I am always 100% of the time happy to get them, but at this point I'm not willing to press the red button and force-push every single PR.  My sincere hope is that Quamina can grow its community a bit and get back to review-required.

But, I still plan to do a lot of coding in the next little while.

**Comment by timbray** [comment]
So, what I'll do, is for every PR, even minor ones, like the current #323, I'll give it 24 hours before pushing.  For longer and more material ones, I'll wait a couple of days at least, to give people an opportunity to provide reviews.

**Reply by embano1** [reply]
+1
